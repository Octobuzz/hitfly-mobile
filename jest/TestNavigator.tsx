// https://github.com/react-navigation/rfcs/issues/75
import React, { Children } from 'react'
import {
  createAppContainer,
  createSwitchNavigator,
  NavigationParams,
} from 'react-navigation'

const TestNavigator = ({
  children,
  params,
}: {
  children: any
  params?: NavigationParams
}) => {
  const Navigator = createAppContainer(
    createSwitchNavigator({
      TestScreen: { screen: () => Children.only(children), params },
    }),
  )
  return <Navigator />
}

export default TestNavigator
