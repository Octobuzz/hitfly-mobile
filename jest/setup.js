import 'jest-styled-components'
import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock'

jest.mock('react-native-vector-icons/Foundation', () => 'Icon')
jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon')
jest.mock('react-native-vector-icons/Feather', () => 'Icon')
jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon')
jest.mock('react-native-vector-icons/SimpleLineIcons', () => 'Icon')

jest.mock('@react-native-community/async-storage', () => mockAsyncStorage)

jest.mock('react-native-track-player', () => {
  return {
    addEventListener: jest.fn(),
    registerEventHandler: jest.fn(),
    registerPlaybackService: jest.fn(),
    setupPlayer: jest.fn(),
    destroy: jest.fn(),
    updateOptions: jest.fn(),
    add: jest.fn(),
    remove: jest.fn(),
    skip: jest.fn(),
    skipToNext: jest.fn(),
    skipToPrevious: jest.fn(),
    removeUpcomingTracks: jest.fn(),
    // playback commands
    reset: jest.fn(),
    play: jest.fn(),
    pause: jest.fn(),
    stop: jest.fn(),
    seekTo: jest.fn(),
    setVolume: jest.fn(),
    setRate: jest.fn(),
    // player getters
    getQueue: jest.fn(),
    getTrack: jest.fn(),
    getCurrentTrack: jest.fn(),
    getVolume: jest.fn(),
    getDuration: jest.fn(),
    getPosition: jest.fn(),
    getBufferedPosition: jest.fn(),
    getState: jest.fn(),
    getRate: jest.fn(),
  }
})

jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native/Libraries/Components/View/View')
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  }
})
