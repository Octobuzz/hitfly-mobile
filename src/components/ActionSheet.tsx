import React, { forwardRef, Ref } from 'react'
import { StyleSheet } from 'react-native'
import {
  ActionSheetCustom,
  ActionSheetCustomProps,
} from 'react-native-actionsheet'
import { getBottomSpace } from 'react-native-iphone-x-helper'
import theme from 'src/theme'

interface Props extends ActionSheetCustomProps {
  // TS говно, не захотел подхватить ref
  ref: Ref<ActionSheetCustom>
}

export type ActionSheetInstance = ActionSheetCustom

const ActionSheet: React.FC<Props> = forwardRef<ActionSheetCustom, Props>(
  (props, ref) => {
    return (
      <ActionSheetCustom
        {...props}
        buttonUnderlayColor={theme.colors.buttonUnderlayColor}
        tintColor={theme.colors.white}
        ref={ref}
        styles={styles}
      />
    )
  },
)

// Для библиотеки так
const styles = StyleSheet.create({
  wrapper: { paddingBottom: getBottomSpace() },
  body: {
    backgroundColor: theme.colors.black,
  },
  messageText: {
    color: theme.colors.white,
    fontSize: 14,
    textAlign: 'center',
  },
  messageBox: {
    height: 50,
    paddingBottom: 0,
    backgroundColor: theme.colors.black,
  },
  buttonBox: {
    height: 50,
    backgroundColor: theme.colors.black,
  },
  buttonText: {
    color: theme.colors.white,
  },
  cancelButtonBox: {
    height: 50,
    marginTop: 6,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.colors.black,
  },
})

export default ActionSheet
