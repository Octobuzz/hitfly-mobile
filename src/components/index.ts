export * from './views'
export * from './fields'
export * from './buttons'

export { default as TextBase } from './TextBase'
export { default as GenreItem } from './GenreItem'
export { default as TextWithLines } from './TextWithLines'
