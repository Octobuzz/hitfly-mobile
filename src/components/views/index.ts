export { default as View } from './View'
export { default as SafeView } from './SafeView'
export { default as Stretcher } from './Stretcher'
export { default as ScrollView } from './ScrollView'
export { default as FormWrapper } from './FormWrapper'
