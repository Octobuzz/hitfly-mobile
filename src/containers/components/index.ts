export * from './SocialAuth'

export { default as SocialAuth } from './SocialAuth'
export { default as HeaderRightButtons } from './HeaderRightButtons'
