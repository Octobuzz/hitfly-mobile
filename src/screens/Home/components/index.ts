export { default as StarsSection } from './StarsSection'
export { default as TracksSection } from './TracksSection'
export { default as SectionHeader } from './SectionHeader'
export { default as SectionWrapper } from './SectionWrapper'
export { default as PlaylistHeader } from './PlaylistHeader'
export { default as CollectionSection } from './CollectionSection'
