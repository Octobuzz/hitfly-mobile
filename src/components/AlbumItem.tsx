import React from 'react'
import { Image } from './Image'
import TextBase from './TextBase'
import { Album, NoAvatarSizeNames } from 'src/apollo'
import { styles } from 'src/constants'
import { useImageSource } from 'src/hooks'
import styled from 'src/styled-components'

const Wrapper = styled.TouchableOpacity`
  width: ${styles.COL2_WIDTH}px;
`

const AlbumImage = styled(Image)`
  height: ${styles.COL2_WIDTH}px;
  width: 100%;
  margin-bottom: 14px;
  border-radius: 4px;
`

const TitleText = styled(TextBase).attrs(() => ({
  numberOfLines: 1,
}))`
  font-family: ${({ theme }) => theme.fonts.medium};
  font-size: 16px;
  line-height: 16px;
  margin-bottom: 8px;
`

const SubTitleText = styled(TextBase).attrs(() => ({
  numberOfLines: 1,
}))`
  font-size: 12px;
  line-height: 12px;
  color: ${({ theme }) => theme.colors.textAlt};
`

interface Props {
  item: Album
  onPress?: (item: Album) => void
}

const AlbumItem: React.FC<Props> = ({ item, onPress }) => {
  const { cover, title, author } = item
  const handlePress = React.useCallback(() => {
    if (onPress) {
      onPress(item)
    }
  }, [onPress, item])

  const source = useImageSource(cover, NoAvatarSizeNames.S_160)

  return (
    <Wrapper onPress={handlePress} testID="albumItem">
      <AlbumImage source={source} />
      <TitleText>{title}</TitleText>
      <SubTitleText>{author}</SubTitleText>
    </Wrapper>
  )
}

export default AlbumItem
