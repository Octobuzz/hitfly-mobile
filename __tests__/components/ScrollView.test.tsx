import React from 'react'
import { render, ReactTestInstanceExtended } from '../../jest/test-utils'
import { ScrollView } from 'src/components/views'

describe('ScrollView', () => {
  const props = {
    paddingTop: 10,
    paddingBottom: 20,
    paddingVertical: 40,
    paddingHorizontal: 10,
  }

  it('renders correctly', () => {
    const { container } = render(<ScrollView />)
    expect(container.children[0] as ReactTestInstanceExtended).toMatchSnapshot()
  })

  it('renders with noFill === false and noPadding === false', () => {
    const { container } = render(
      <ScrollView noFill={false} noPadding={false} {...props} />,
    )
    expect(container.children[0] as ReactTestInstanceExtended).toMatchSnapshot()
  })
  it('renders with noFill === true and noPadding === true', () => {
    const { container } = render(<ScrollView noFill noPadding {...props} />)
    expect(container.children[0] as ReactTestInstanceExtended).toMatchSnapshot()
  })
  it('renders with no vertical and horizontal padding', () => {
    const { container } = render(
      <ScrollView
        noVerticalPadding={false}
        noHorizontalPadding={false}
        {...props}
      />,
    )
    expect(container.children[0] as ReactTestInstanceExtended).toMatchSnapshot()
  })
  it('renders with vertical and horizontal padding', () => {
    const { container } = render(
      <ScrollView noVerticalPadding noHorizontalPadding {...props} />,
    )
    expect(container.children[0] as ReactTestInstanceExtended).toMatchSnapshot()
  })
  it('renders with addBottomSafePadding === false', () => {
    const { container } = render(
      <ScrollView addBottomSafePadding={false} {...props} />,
    )
    expect(container.children[0] as ReactTestInstanceExtended).toMatchSnapshot()
  })
  it('renders with addBottomSafePadding === true', () => {
    const { container } = render(<ScrollView addBottomSafePadding {...props} />)
    expect(container.children[0] as ReactTestInstanceExtended).toMatchSnapshot()
  })
})
