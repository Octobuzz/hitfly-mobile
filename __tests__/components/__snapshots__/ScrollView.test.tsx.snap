// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollView renders correctly 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 16,
      "paddingHorizontal": 16,
      "paddingTop": 16,
      "paddingVertical": 16,
    }
  }
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;

exports[`ScrollView renders with addBottomSafePadding === false 1`] = `
<ScrollView
  addBottomSafePadding={false}
  contentContainerStyle={
    Object {
      "paddingBottom": 20,
      "paddingHorizontal": 10,
      "paddingTop": 10,
      "paddingVertical": 16,
    }
  }
  paddingBottom={20}
  paddingHorizontal={10}
  paddingTop={10}
  paddingVertical={40}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;

exports[`ScrollView renders with addBottomSafePadding === true 1`] = `
<ScrollView
  addBottomSafePadding={true}
  contentContainerStyle={
    Object {
      "paddingBottom": 20,
      "paddingHorizontal": 10,
      "paddingTop": 10,
      "paddingVertical": 16,
    }
  }
  paddingBottom={20}
  paddingHorizontal={10}
  paddingTop={10}
  paddingVertical={40}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;

exports[`ScrollView renders with no vertical and horizontal padding 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 20,
      "paddingHorizontal": 10,
      "paddingTop": 10,
      "paddingVertical": 16,
    }
  }
  noHorizontalPadding={false}
  noVerticalPadding={false}
  paddingBottom={20}
  paddingHorizontal={10}
  paddingTop={10}
  paddingVertical={40}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;

exports[`ScrollView renders with noFill === false and noPadding === false 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 20,
      "paddingHorizontal": 10,
      "paddingTop": 10,
      "paddingVertical": 16,
    }
  }
  noFill={false}
  noPadding={false}
  paddingBottom={20}
  paddingHorizontal={10}
  paddingTop={10}
  paddingVertical={40}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;

exports[`ScrollView renders with noFill === true and noPadding === true 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 0,
      "paddingHorizontal": 0,
      "paddingTop": 0,
      "paddingVertical": 0,
    }
  }
  noFill={true}
  noPadding={true}
  paddingBottom={20}
  paddingHorizontal={10}
  paddingTop={10}
  paddingVertical={40}
  style={
    Array [
      Object {},
    ]
  }
/>
`;

exports[`ScrollView renders with vertical and horizontal padding 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 0,
      "paddingHorizontal": 0,
      "paddingTop": 0,
      "paddingVertical": 0,
    }
  }
  noHorizontalPadding={true}
  noVerticalPadding={true}
  paddingBottom={20}
  paddingHorizontal={10}
  paddingTop={10}
  paddingVertical={40}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;
